donation/models/partner.py

```diff
@@ -12,11 +12,15 @@ class ResPartner(models.Model):
     @api.depends('donation_ids.partner_id')
     def _compute_donation_count(self):
         # The current user may not have access rights for donations
-        for partner in self:
-            try:
-                partner.donation_count = len(partner.donation_ids)
-            except Exception:
-                partner.donation_count = 0
+        try:
```

Comment:
Why do you enclose it in a try?

Reply:
As I say in the comment above: "The current user may not have access rights on donation.donation".

Reply:
But this is not the solution IMO. Put `compute_sudo=True` in the field definition, and filter by company. The smartbutton should be also only visible to those with permissions, and putting the groups at view level, they won't be computed when no access.